src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -967,10 +969,25 @@ def _infer_schema_for_file_format(
         drop_tmp_file_format_if_exists_query: Optional[str] = None
         use_temp_file_format = "FORMAT_NAME" not in self._cur_options
         file_format_name = self._cur_options.get("FORMAT_NAME", temp_file_format_name)
-        infer_schema_options = self._cur_options.get("INFER_SCHEMA_OPTIONS", None)
+        infer_schema_options = self._cur_options.get("INFER_SCHEMA_OPTIONS", {})
+
+        # When pattern is set we should only consider files that match the pattern during schema inference
+        # If no files match fallback to trying to read all files.
+        if (
+            pattern := self._cur_options.get("PATTERN", None)
+        ) and "FILES" not in infer_schema_options:
+            matches = self._session._conn.run_query(
+                f"list {path} pattern = '{pattern}'"
+            )["data"]
+            if len(matches):
+                infer_schema_options["FILES"] = tuple(
+                    m[0].split("/")[-1] for m in matches
+                )
```

Comment:
can we add a comment on what the return value is and what the tuples will look like